Services.Tests/ExternalDataProviders/GitlabDataSourceAdapteeTest.cs

```diff
@@ -0,0 +1,417 @@
+/*
+* Digital Excellence Copyright (C) 2020 Brend Smits
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published
+* by the Free Software Foundation version 3 of the License.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty
+* of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+* See the GNU Lesser General Public License for more details.
+*
+* You can find a copy of the GNU Lesser General Public License
+* along with this program, in the LICENSE.md file in the root project directory.
+* If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
+*/
+
+using FluentAssertions;
+using Microsoft.Extensions.Configuration;
+using Moq;
+using NUnit.Framework;
+using Services.ExternalDataProviders;
+using Services.ExternalDataProviders.Resources;
+using Services.Tests.Base;
+using Services.Tests.ExternalDataProviders.DataSources.Gitlab;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Services.Tests.ExternalDataProviders
+{
+
+    /// <summary>
+    ///     GitlabDataSourceAdapteeTest
+    /// </summary>
+    /// <seealso cref="IGitlabDataSourceAdaptee" />
+    [TestFixture]
+    public class GitlabDataSourceAdapteeTest : AdapteeTest<IGitlabDataSourceAdaptee>
```

Comment:
The following methods in GitlabDataSourceAdaptee.cs are not covered:

GetTokens
GetAllProjects
GetProjectById
GetAllPublicProjects
GetPublicProjectFromUri
GetPublicProjectById